ArthurAtangana/SYSC3303A_Project
# Messaging module refactor (Iteration 2)

## What type of PR is this? (check all applicable)


## Description
Made interfaces in networking into abstract classes for code reuse.
- Added binding methods in transmitter.
- Reworked receiver to use "dequeueMessage()" (because this is what the subsystems are concerned with), subclasses implement their own way to receive opaque to subsystems. 
- Raised most of ReceiverDMA code to Receiver class since most of it will be re-usable.
- Updated main and other classes to new interfaces
- Also renamed a lot of the structuring in networking, it will grow in size soon so this seemed like the right time to do it (first two commits are related to this, I'd suggest going commit by commit for easier review).
- Note: some docs might not be fully done, feel free to add technical debt comments

## How to test
Run main, should behave as before when iteration 2 was submitted.
Run networking tests, should all pass.

## Added/updated tests?

- [X] Yes: New slightly more detailed ReceiverDMA tests, also renaming stuff

## Screenshots
